diff_drive_controller/test/diffbot.cpp

```diff
@@ -40,19 +41,54 @@ int main(int argc, char **argv)
   ros::init(argc, argv, "diffbot");
   ros::NodeHandle nh;
 
+  // This should be set in launch files
+  // as well
+  nh.setParam("/use_sim_time", true);
+
   Diffbot<> robot;
   ROS_WARN_STREAM("period: " << robot.getPeriod().toSec());
   controller_manager::ControllerManager cm(&robot, nh);
 
-  ros::Rate rate(1.0 / robot.getPeriod().toSec());
+  ros::Publisher clock_publisher = nh.advertise<rosgraph_msgs::Clock>("/clock", 1);
```

Comment:
The simulation now publishes its own clock which should fix time/integration erratic failures.